Lib/fontTools/ttLib/tables/otBase.py

```diff
@@ -294,6 +307,19 @@ def getData(self):
 
 		return bytesjoin(items)
 
+	def getDataForHarfbuzz(self):
+		"""Assemble the data for this writer/table with all offset field set to 0"""
+		items = list(self.items)
+		for i, item in enumerate(items):
+			if hasattr(item, "getData"):
+				#offset value is not needed in harfbuzz repacker, so setting offset to 0 to avoid overflow here
+				if item.offsetSize == 4 or item.offsetSize == 2 or item.offsetSize == 3:
+					items[i] = {2: packUShort, 3: packUInt24, 4: packULong}[item.offsetSize](0)
```

Comment:
What about you define offsetSizes dict outside of the loop, something like:
```
offsetSizes = {2: packUShort, 3: packUInt24, 4: packULong}
for i, item in enumerate(items):
   ...
   if item.offsetSize in offsetSizes:
   ...
   else:
   ...
```

This is probably more efficient.